File: L03-Time-Complexity/FrogJmp.java
// Date: Nov 26, 2021
// https://app.codility.com/programmers/lessons/3-time_complexity/frog_jmp/

// time complexity: O(1)
// space complexity: O(1)
// result: https://app.codility.com/demo/results/trainingU8JAWS-BF4/
public int solution(int X, int Y, int D) {
	// return ((Y - X - 1) / D) + 1;
	return (int) Math.ceil(1.0 * (Y - X) / D);
}

File: L03-Time-Complexity/FrogJmp.swift
// Input: X = 10, Y = 85, D = 30
// Output: 3

// result: https://app.codility.com/demo/results/trainingY8JU2H-WKM/

public func solution(_ X : Int, _ Y : Int, _ D : Int) -> Int {
    var ans = Double(Y - X) / Double(D)
    ans.round(.up)
    return Int(ans)
}
